db_controller.py

```diff
@@ -0,0 +1,36 @@
+import logging
+from sqlite3 import Error
+import sqlite3
+
+
+def create_connection(db_file):
+    """ create a database connection to a SQLite database """
+    conn = None
+    try:
+        conn = sqlite3.connect(db_file)
+    except Error as e:
+        logging.getLogger('DB create').error(e)
+    return conn
+
+
+def create_table(conn):
+    query = """CREATE TABLE IF NOT EXISTS links (link text NOT NULL UNIQUE);"""
```

Comment:
Consider using an ORM like `sqlalchemy`(3rd party) instead of raw SQL.  This is highly-subjective, but I think ORM's are much cleaner.